### PR TITLE
[15.0] [IMP] improve tests performance

### PR DESCRIPTION
**Pull request by josep-tecnativa**
CC @Tecnativa TT45458

### CODE REVIEW

**Comment by josep-tecnativa** [comment]
ping @pedrobaeza

**Comment by pedrobaeza** [comment]
Typo in commits. Please change it everywhere: permformance > performance

File: account_financial_report/tests/test_aged_partner_balance.py [hunk 1]
```diff
@@ -8,6 +8,17 @@
 class TestAgedPartnerBalance(TransactionCase):
     def setUp(self):
         super().setUp()
+        self.env = self.env(
```
Comment (pedrobaeza) [inline account_financial_report/tests/test_aged_partner_balance.py]: Change it to setUpClass (with its implications to also get better performance)

File: account_tax_balance/tests/test_account_tax_balance.py [hunk 2]
```diff
@@ -19,7 +19,16 @@
 class TestAccountTaxBalance(HttpCase):
     def setUp(self):
         super().setUp()
-
+        self.env = self.env(
```
Comment (pedrobaeza) [inline account_tax_balance/tests/test_account_tax_balance.py]: The same.

File: partner_statement/tests/test_activity_statement.py [hunk 3]
```diff
@@ -10,26 +10,36 @@
...
-        self.statement_model = self.env["report.partner_statement.activity_statement"]
-        self.wiz = self.env["activity.statement.wizard"]
-        self.report_name = "partner_statement.activity_statement"
-        self.report_name_xlsx = "p_s.report_activity_statement_xlsx"
-        self.report_title = "Activity Statement"
-        self.today = fields.Date.context_today(self.wiz)
+        cls.partner1 = cls.env.ref("base.res_partner_1")
+        cls.partner2 = cls.env.ref("base.res_partner_2")
+        cls.g_account_user = cls.env.ref("account.group_account_user")
+
```
Comment (pedrobaeza) [inline partner_statement/tests/test_activity_statement.py]: Take the occasion to remove empty lines inside the method.

File: partner_statement/tests/test_activity_statement.py [hunk 4]
```diff
@@ -10,26 +10,36 @@
...
+        cls.user = cls._create_user("user_1", [cls.g_account_user], cls.company).id
```
Comment (pedrobaeza) [inline partner_statement/tests/test_activity_statement.py]: Remove the method `_create_user` and use:

```
from odoo.tests import new_test_user
```

and call it this way:

```
cls.user = new_test_user(cls.env, login="user_1", groups="account.group_account_user")
```

Do the same for each of the custom "user creation".

**Comment by pedrobaeza** [comment]
The typo `permformance` (instead of `performance`) still exists in the commit names.

**Review by pedrobaeza — APPROVED** [review]
Thanks! Let's save some CPU time with this. Fast-tracking as straight-forward:

/ocabot merge patch

**Comment by pedrobaeza** [comment]
Please do the same for 16.0 (and the rest of involved repos).

**Pull request merged**